Gateway/Command/FetchTransactionCommand.php

```diff
@@ -71,16 +80,19 @@ public function execute(array $commandSubject)
     {
         /** @var \Magento\Sales\Model\Order\Payment $payment */
         $payment = $commandSubject['payment']->getPayment();
-        $result = $this->api->getTransaction(
-            $commandSubject['transactionId'],
-            $payment->getOrder()->getStoreId()
-        );
+        $transactionId = $commandSubject['transactionId'] ?? null;
+
+        if ($payment->getTransactionId()) {
+            $transactionId = $this->repository->get($transactionId)->getTxnId();
+        }
+
+        $result = $this->api->getTransaction($transactionId, $payment->getOrder()->getStoreId());
```

Comment:
Can you explain what happens here?

Seems we can end up in a situation where `$transactionId` is `null` when calling:

```php
$this->api->getTransaction($transactionId, $payment->getOrder()->getStoreId());
```

In what scenarios will `$commandSubject['transactionId']` be null? Is that the query parameters from the callback?

Reply:
@andersem , it should never be null, at least in native Magento. This check will help to avoid fatal error if somehow it is null. For example third party extension overriding that param earlier in the code... so it is just a measure of additional safety.

In general the issue is that Magento behaves strange and in case of fetch transaction information which is triggered from admin panel Magento passes Dintero transaction id into `$subject['transactionId']`, however, during invoice capture `$subject['transactionId']` contains numeric value of transaction id record  (primary key of `sales_payment_transaction` table)in Magento database as a result the fetch fails.

Reply:
@chaikovskyia All right, thanks for the explanation. I'm adding it to the PR description. :)